### PR TITLE
Fix issues with verification for fallback type

### LINKED ISSUE #180: Fix issues with verification of versioning fallback type
#### Description:

Two issues with the check.

FIrst, namespace grouping need an exception for Adapters namespace, as they require both third and fourth namespace key to be identifiable.

Second is the StartsWith check, meant to handle generics, that is to broad and is catching to much. To be changed to splitting the name by '`' instead.

#### Steps to reproduce:

Testscript will be provided on the PR.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #180

Fixing two issues with the fallback type
- Fixes the grouping by namespace to make sure adapter namespaces are appropriately grouped
- Changing the checking of exact match or StartsWith to instead splitting by generics sign and only check for exact match

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EjFXfSKJ2mlNmQMbHWWXH4sBwNWOSNPgpdK3uhRk4v1jfw?e=ZcswlC

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
@BHoMBot check versioning

**Comment by IsakNaslundBh** [comment]
@BHoMBot check versioning

**Review by adecler — APPROVED** [review]
I am happy with the two corrections.

The changes observed is the test results are making sense so also happy with merging this PR as long as we action the fix for the methods that are now failing.

image

**Comment by IsakNaslundBh** [comment]
@BHoMBot check copyright-compliance
@BHoMBot check core
@BHoMBot check installer
@BHoMBot check ready-to-merge

**Review by FraserGreenroyd — APPROVED** [review]
I would concur with @adecler .

With methods now being picked up as potential failures with this change, merging this will start to report blocks on PRs. When we go for merge therefore we need to make sure the new reports of failures are handled.

We might thus wish to hold off merging until Monday as today is the final merge day of the milestone for features, so blocking things today for pre-existing issues may not be wise. But am also happy to merge now and give people that little kick to get it fixed 😉

**Comment by IsakNaslundBh** [comment]
Thanks both.

To be clear what I meant with "failure" in the tescript, it means that those methods would fail the versioning check if the are deleted without any other versioning measures. For example, if the `FoundationBar` methods are deleted with main setup of the check, they will report a pass. After this PR, it will be reported as a fail. 

I personally do not see any real issues with this, as the previous passes where wrong anyway. We also have over 200 create methods that are failing the check for both options as it is. What this PR is doing is just slightly modifying the failing lists with the methods from the snippet shown by @adecler , hence not sure if anything needs to be fixed per se. Happy to hold off merging until next week to avoid any risk for merging things today though.

**Comment by FraserGreenroyd** [comment]
Thanks @IsakNaslundBh that makes sense, in which case I'm happy for merge today if you are on that basis 😄

**Comment by adecler** [comment]
> To be clear what I meant with "failure" in the tescript, it means that those methods would fail the versioning check if the are deleted without any other versioning measures. For example, if the `FoundationBar` methods are deleted with main setup of the check, they will report a pass. After this PR, it will be reported as a fail.

Understood. I would still raise issues to action the fixes while it is fresh and clear in our mind.

**Pull request merged**
